src/main/java/org/jenkinsci/maven/plugins/hpi/ValidateMojo.java

```diff
@@ -1,24 +1,33 @@
 package org.jenkinsci.maven.plugins.hpi;
 
+import com.google.common.collect.Sets;
 import hudson.util.VersionNumber;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
 
 /**
  * Make sure that we are running in the right environment.
  *
  * @author Kohsuke Kawaguchi
  */
-@Mojo(name = "validate", defaultPhase = LifecyclePhase.VALIDATE)
-public class ValidateMojo extends AbstractJenkinsMojo {
+@Mojo(name = "validate", defaultPhase = LifecyclePhase.VALIDATE, requiresDependencyResolution = ResolutionScope.RUNTIME)
+public class ValidateMojo extends AbstractHpiMojo {
```

Comment:
It will prevent from using the mojo outside the HPI packaging use-cases. I would recommend to retain compatibility and to implement a new mojo instead.